### PR TITLE
diag(bitnet): compare dumped logit ranks

### PR DESCRIPTION
**Pull request by EffortlessSteven**
## Summary

Preserves dumped CLI `softmax_rank` values in `bitnet-reference-compare` output for top and selected/probe logits.

This keeps the reference-server probability probes, Rust CPU logits, and strict A770 logits in one diagnostic receipt with token pieces, probabilities, and ranks side by side.

## Evidence

Validated locally:

```text
cargo test --locked -p xtask --no-default-features --bin xtask bitnet_reference_compare -- --nocapture
rustfmt --check --edition 2024 --config skip_children=true xtask\src\bitnet_reference_compare.rs
git diff --check
cargo run --locked -p xtask --no-default-features -- bitnet-reference-compare --reference target/a770-diagnostic/bitnet-reference-run-gguf-tokenizer-fix.json --reference-server target/a770-diagnostic/bitnet-reference-server-run-first-token.json --cpu target/a770-diagnostic/reference-plan-cpu-first-token.json --a770 target/a770-diagnostic/reference-plan-a770-first-token.json --output target/a770-diagnostic/bitnet-reference-compare-with-ranks.json --format json
```

Real diagnostic result from the regenerated compare receipt:

```text
Reference server selected content: 2
Rust CPU token 2: softmax_probability≈0.000060, softmax_rank=1173
Strict A770 token 2: softmax_probability≈0.000058, softmax_rank=1218
Rust CPU/A770 top token: .ps, softmax_rank=1
```

## Not claims

This PR is diagnostic-only. It does not claim:

```text
reference generated token parity
A770 semantic quality
A770 performance
selected_attention_residency
resident_kv_decode
attention_scores_residency
softmax_residency
attention_value_mix_residency
full_support_op_residency
full_device_residency
completion
```

### CODE REVIEW

**Comment by EffortlessSteven** [comment]
PR: #4875
Lane: BitNet/A770 diagnostic lineage
Disposition: closed as superseded
Base/head: diagnostic branch-chain head is an ancestor of merged #5354 (770/diag-rust-score-input-operand-drift); later branch-chain evidence continued through #5722/#5725.
Claim boundary: diagnostic-only; no A770/OpenCL/CUDA/AVX/server/speed/residency/reference-parity claim promoted.
Generated files: none reviewed for mainline merge.
Proof commands: lineage containment via git merge-base against origin/a770/diag-rust-score-input-operand-drift; no merge proof run because this is a closure disposition.
CI status: not used as merge proof.
Related PRs: #5354, #5722, #5725
Supersedes: none
Blocked by: none
Review decision: close; transient diagnostic probe captured by newer branch-chain evidence. Durable diagnostic value should be recovered from the merged summary chain rather than landing this stepping-stone PR on main.
